src/backend/api/meals.js

```diff
@@ -1,16 +1,165 @@
-const express = require("express");
+const express = require("express")
 const router = express.Router();
 const knex = require("../database");
 
+
+
+
+
+//  /api/meals/:meal_id/reviews	GET	Returns all reviews for a specific meal.
+//  Reviews
+
+router.get("/:meal_id/reviews", async (request, response) => {
+
+
+  let mealReviews = await knex("Review").select().where({ "Review.meal_id": request.params.meal_id })
+
+
+  if (mealReviews.length > 0) {
+    response.json(mealReviews)
+  } else {
+    response.status(404).json({ error: "Review Not found" })
+  }
+})
+
+
+
+
 router.get("/", async (request, response) => {
-  try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const meal = await knex("meals").select();
-    response.json(meal);
 
-  } catch (error) {
-    throw error;
+  // maxPrice	Number	Returns all meals that are cheaper than maxPrice.	api/meals?maxPrice=90
+
+  if ("maxPrice" in request.query) {
+    if (isNaN(request.query.maxPrice)) {
+
+      response.status(404).json({ error: "max price is not a number" });
+
+    } else {
+      const maxPriceMeals = await knex("meals").select().where('meals.price', '<', `${request.query.maxPrice}`);
+
+      response.json(maxPriceMeals);
```

Comment:
You still have the problem of sending a response for each parameter. If I provide both maxPrice and title for instance what would happen? This route would try to send 2 responses but that's not possible, only the first response will be sent.

The high-level approach is that you need to construct the query as you and then at the end of this function, you send the query to the database and then send a **single response**.

Reply:
if I return  response.json(maxPriceMeals) on line 41 (end of the function), does that avoid sending two responses?

Reply:
Yes it avoids it but it also means that you then ignore other potential parameters, for instance if the `title` parameter is set.